docs/developer_setup.md

```diff
@@ -0,0 +1,41 @@
+## Setting up a Holochain development environment
+
+### Pre-requisites
+
+You will need one of the following
+
+- *recommended* - Having `nix` installed following these [instructions](https://nixos.org/download.html).
+- *alternative* - Having Rust installed via [rustup](https://www.rust-lang.org/tools/install) and configured to use the stable toolchain.
+
+### Build holochain
+
+If using Nix, start by opening a developer shell using `nix develop .#coreDev` at the root of the repository. This will take
+some time you run it.
+
+Now you can build the project with `cargo build`.
+
+This is a good check that your environment is ready to use. If you have problems here you can [get in touch](https://github.com/holochain/holochain/blob/develop/CONTRIBUTING.md#coordination) or open an issue.
+
+### Run the tests
+
+There is a [testing guide](https://github.com/holochain/holochain/blob/develop/docs/core_testing.md) which will get you started running
+the tests the same way the CI does.
+
+### Build CLI tools
+
+Other documentation will assume you have access to CLI tools provided by this repository, so it's a good idea to build these now.
```

Comment:
Why are these binaries built here? If the developer is working on Holochain itself, they shouldn't need to build them.

Or are you adding this section as a preparation for the next step?

Reply:
Exactly that, if you're wanting to use the sandbox or interact with a running conductor then having these built is useful. I'll update the section header to make it clearer why this is being done, so readers can decide if they want to do it or not